### PR TITLE
Issue #143 Table of Contents

### PR DESCRIPTION
**Pull request by igor-baiborodine**
Implementation of the table of contents feature including the fix for the #252 issue.

Test website: https://bilberry-toc-test.netlify.app/

### CODE REVIEW

File: i18n/en.toml [hunk 1]
```diff
@@ -35,3 +35,6 @@ other = "Other languages"
 one = "1 min read"
 other = "{{.Count}} min read"
 
+[tableOfContents]
+other = "Table of Contents"
```
Comment (Lednerb) [inline i18n/en.toml:39]: Does hugo produce a build error if this string is missing?
If so, we need to add the translations for all supported languages so we don't introduce a breaking change here that woud cause a versioning step to version 3 of this theme.

I would like to have it backwards compatible, so users can upgrade the theme without interruptions.

**Review by Lednerb — CHANGES_REQUESTED** [review]

File: assets/sass/_articles.scss [hunk 2]
```diff
@@ -160,6 +160,51 @@ article {
...
+                        font-size: 95%;
+
+                        ul {
+                            display: inherit;
+                            padding-left: 0.9rem;
+                            font-size: 95%;
+                        }
+                    }
+                }
+            }
```
[the remark this answers is not in the document]
Comment (igor-baiborodine) [reply assets/sass/_articles.scss]: I simplified it to this:
`ul {
                &:not(:first-child) {
                    display: list-item;
                    padding-left: 0.9rem;
                    font-size: 95%;
                }
            }`

File: i18n/en.toml (hunk 1 above)
Comment (igor-baiborodine) [reply i18n/en.toml:39]: I piped with `default "tableOfContents' to display the code if a corresponding i18n is not implemented:

`<h2>{{ i18n "tableOfContents" | default "tableOfContents" }}</h2>`

File: assets/sass/_articles.scss [hunk 3]
```diff
@@ -160,6 +160,41 @@ article {
...
+                padding-left: 0;
+            }
+
+            ul {
+                &:not(:first-child) {
+                    display: list-item;
+                    padding-left: 0.9rem;
+                    font-size: 95%;
+                }
+            }
```
Comment (Lednerb) [inline assets/sass/_articles.scss:180]: Please combine these blocks into one and I will merge the PR :+1: 

Thanks for your great contribution again!

**Review by Lednerb — CHANGES_REQUESTED** [review]

**Review by Lednerb — APPROVED** [review]

**Pull request merged**